src/prefect/_internal/retries.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from functools import wraps
+from typing import Any, Callable, Optional, Tuple, Type
+
+from prefect.logging.loggers import get_logger
+from prefect.utilities.math import clamped_poisson_interval
+
+logger = get_logger("retries")
+
+
+def exponential_backoff_with_jitter(
+    attempt: int, base_delay: float, max_delay: float
+) -> float:
+    average_interval = min(base_delay * (2**attempt), max_delay)
+    return clamped_poisson_interval(average_interval, clamping_factor=0.3)
+
+
+def retry_async_fn(
+    max_attempts: int = 3,
+    backoff_strategy: Optional[Callable[[int, float, float], float]] = None,
+    base_delay: float = 1,
+    max_delay: float = 10,
+    retry_on_exceptions: Optional[Tuple[Type[Exception], ...]] = None,
```

Comment:
I think we should be ok to use a tuple as the default value here since tuples are immutable.